### DISCUSSION TITLE
Debug logging for a custom linter

### DISCUSSION THREAD

**Discussion by prologic** [golangci/golangci-lint#3581] [Q&A] [answered]
How do you do this?

The [docs](https://golangci-lint.run/contributing/debug/) on this are basically not very helpful at all and point to [source code](https://github.com/golangci/golangci-lint/blob/master/pkg/logutils/logutils.go) for the different values that `GL_DEBUG` can have.

The PR(s) I saw that "improved documentation" on this, actually made it worse. I need to know one simple thing:

- If I `log.Printf()` or `fmt.Fprintf(os.Stderr, ...)` in my linter, where do these logs go?

I don't really want to dig into the source code to have to find this out 😆

Thank you! 🙏

Things I have tried are passing `GL_DEBUG` with values like `env,runner,loader` and even `env,loader,runnner,<mylinter>` with no success.

**Comment by krish7919** [comment]
I am interested in this too. Are there any good linter examples to refer to?

**Comment by ldez** [comment, marked as answer]
You should use `GL_DEBUG=linters_output` to output logs from linters.
